parts/dcosprovision.sh

```diff
@@ -4,11 +4,31 @@ MESOSDIR=/var/lib/mesos/dl
 mkdir $MESOSDIR
 
 # load the env vars
-curl -fLsSv --retry 20 -Y 100000 -y 60 -o $MESOSDIR/d.deb https://az837203.vo.msecnd.net/dcos-deps/docker-engine_1.11.2-0~xenial_amd64.deb &
-curl -fLsSv --retry 20 -Y 100000 -y 60 -o $MESOSDIR/1.deb https://az837203.vo.msecnd.net/dcos-deps/libipset3_6.29-1_amd64.deb &
-curl -fLsSv --retry 20 -Y 100000 -y 60 -o $MESOSDIR/2.deb https://az837203.vo.msecnd.net/dcos-deps/ipset_6.29-1_amd64.deb &
-curl -fLsSv --retry 20 -Y 100000 -y 60 -o $MESOSDIR/3.deb https://az837203.vo.msecnd.net/dcos-deps/unzip_6.0-20ubuntu1_amd64.deb &
-curl -fLsSv --retry 20 -Y 100000 -y 60 -o $MESOSDIR/4.deb https://az837203.vo.msecnd.net/dcos-deps/libltdl7_2.4.6-0.1_amd64.deb &
+. /etc/mesosphere/setup-flags/dcos-deploy-environment
+
+# default dc/os component download address (Azure CDN)
+DOCKER_ENGINE_DOWNLOAD_URL=https://az837203.vo.msecnd.net/dcos-deps/docker-engine_1.11.2-0~xenial_amd64.deb
+LIBIPSET_DOWNLOAD_URL=https://az837203.vo.msecnd.net/dcos-deps/libipset3_6.29-1_amd64.deb
+IPSET_DOWNLOAD_URL=https://az837203.vo.msecnd.net/dcos-deps/ipset_6.29-1_amd64.deb
+UNZIP_DOWNLOAD_URL=https://az837203.vo.msecnd.net/dcos-deps/unzip_6.0-20ubuntu1_amd64.deb
+LIBLTDL_DOWNLOAD_URL=https://az837203.vo.msecnd.net/dcos-deps/libltdl7_2.4.6-0.1_amd64.deb
+
+case $DCOS_ENVIRONMENT in
+    # because of Chinese GreatWall Firewall, the default packages on Azure CDN is blocked. So the following Chinese local mirror url should be used instead.
+    AzureChinaCloud)
+        DOCKER_ENGINE_DOWNLOAD_URL=http://acsengine.blob.core.chinacloudapi.cn/dcos/docker-engine_1.11.2-0~xenial_amd64.deb

```

Comment:
For bootstrap download url I put it in the internal model. As for these urls I followed current design of this shell script to determine the url based on the env. Put these urls into internal model and dynamically generate them using parameters may introduce more configurations. I think it may be better to put them in current location. @colemickens do you think it make sense?